python/services/dataservice/dmod/dataservice/initial_data_adder_impl.py

```diff
@@ -223,13 +231,15 @@ def add_initial_data(self):
         DmodRuntimeError
             Raised when initial data could not be assembled and/or added successfully to the dataset.
         """
+        original_domain = self._dataset_manager.datasets[self._dataset_name].data_domain
+
         if self.partial_realization_config is not None:
             raise DmodRuntimeError(f"{self.__class__.__name__} can't have 'None' for partial realization property")
 
         try:
             real_config: NgenRealization = self.build_realization_config_from_partial()
             if not self._dataset_manager.add_data(dataset_name=self._dataset_name, dest=self._REAL_CONFIG_FILE_NAME,
-                                                  data=json.dumps(real_config.json()).encode()):
+                                                  data=json.dumps(real_config.json()).encode(), domain=original_domain):
```

Comment:
I think this should be, no?

```suggestion
                                                  data=real_config.json().encode(), domain=original_domain):
```

```python
a = real_config.json().encode()
b = json.dumps(real_config.json()).encode()

assert type(json.loads(a)) == dict
assert type(json.loads(b)) == str
```

Reply:
Good catch.  I've pushed a change for it.